### DISCUSSION TITLE
Why `Set-Cookie` headers are not returned with server/api route?

### DISCUSSION THREAD

**Discussion by crotanite** [nuxt/nuxt#17148] [Questions] [answered]
Does anyone know why when making a request via Nuxt's server/api routes "Set-Cookie" headers are NOT returned, but when doing it with the full url by the useFetch method they are?

*server/api/[...].ts*
```js
const config = useRuntimeConfig()
const baseURL = config.apiUrl

export default defineEventHandler(async event => {
    const method = useMethod(event)
    const params = useQuery(event)
    const body = method === "GET" ? undefined : await useBody(event)
    const url = event.req.url.replace('api/', '')

    return await $fetch(url, {
        baseURL,
        credentials: 'include',
        method,
        params,
        body
    })
})
```

e.g.
If I make a request like so: `useFetch('http://api.mysite.test/csrf-cookie')` I get the set-cookie headers but if I make one like `useFetch('/api/csrf-cookie')` which leads to the above code, they aren't?

**Comment by danielroe** [comment, marked as answer]
This might explain it: https://v3.nuxtjs.org/guide/features/data-fetching/#isomorphic-fetch-and-fetch

**Comment by nathanchase** [comment]
@crotanite Yeah, like @danielroe said, you'll have to use `useRequestHeaders` if you want to pass the headers. 

https://v3.nuxtjs.org/api/composables/use-request-headers/ could use a bit more specific documentation, though, for this use case of sending headers to server API handlers.

**Comment by crotanite** [comment]
Thanks guys, yeah I think I must have just missed this in the docs.
Honestly though I found this is just too much of a pain to implement and switched away from it.

For example, from Laravel I get two cookies when hitting the csrf-cookie route, but trying to split them and set them is just overly complicated. Not worth the extra work for me.
